src/network/mcpe/handler/LoginPacketHandler.php

```diff
@@ -70,19 +72,33 @@ public function handleLogin(LoginPacket $packet) : bool{
 
 		$clientData = $this->parseClientData($packet->clientDataJwt);
 
-		try{
-			$skin = $this->session->getTypeConverter()->getSkinAdapter()->fromSkinData(ClientDataToSkinDataHelper::fromClientData($clientData));
-		}catch(\InvalidArgumentException | InvalidSkinException $e){
-			$this->session->disconnectWithError(
-				reason: "Invalid skin: " . $e->getMessage(),
-				disconnectScreenMessage: KnownTranslationFactory::disconnectionScreen_invalidSkin()
-			);
+		$this->server->getAsyncPool()->submitTask(new ProcessSkinTask(
+			ClientDataToSkinDataHelper::fromClientData($clientData),
+			function(?Skin $skin, ?string $error) use ($packet, $clientData, $extraData){
```

Comment:
Session might not be connected anymore here

Reply:
```php
if(!$this->session->isConnected()){
	return;
}
```

is added for connection state check